src/libraries/Common/src/System/Runtime/InteropServices/ComEventsMethod.cs

```diff
@@ -220,6 +225,40 @@ public void RemoveDelegate(Delegate d)
             }
         }
 
+        public void RemoveDelegates(Func<Delegate, bool> condition)
+        {
+            lock (_delegateWrappers)
+            {
+                // Find delegate wrapper indexes
+                List<int> toRemove = new List<int>();
+                for (int i = 0; i < _delegateWrappers.Count; i++)
+                {
+                    DelegateWrapper wrapper = _delegateWrappers[i];
+                    Delegate[] invocationList = wrapper.Delegate.GetInvocationList();
+                    foreach (Delegate delegateMaybe in invocationList)
+                    {
+                        if (condition(delegateMaybe))
+                        {
+                            Delegate? newDelegate = Delegate.Remove(wrapper!.Delegate, delegateMaybe);
+                            if (newDelegate != null)
+                            {
+                                wrapper.Delegate = newDelegate;
+                            }
+                            else
+                            {
+                                toRemove.Add(i);
+                            }
+                        }
+                    }
+                }
+
+                foreach (int idx in toRemove)
+                {
+                    _delegateWrappers.RemoveAt(idx);
+                }
```

Comment:
Do we need to remove higher indices before lower?

Reply:
Ack - yes.